hazelcast/src/main/java/com/hazelcast/jet/impl/JobClassLoaderService.java

```diff
@@ -0,0 +1,311 @@
+        logger.fine("Create processor classloader map for job " + idToString(jobId));
+        String customLibDir = nodeEngine.getProperties().getString(ClusterProperty.PROCESSOR_CUSTOM_LIB_DIR);
+        Map<String, ClassLoader> classLoaderMap = new HashMap<>();
+        for (Entry<String, List<String>> entry : jobConfig.getCustomClassPaths().entrySet()) {
+            List<URL> list = entry.getValue().stream()
+                                  .map(jar -> {
+                                      try {
+                                          Path path = Paths.get(customLibDir, jar);
+                                          return path.toUri().toURL();
+                                      } catch (MalformedURLException e) {
+                                          throw new JetException(e);
+                                      }
+                                  })
+                                  .collect(Collectors.toList());
+            URL[] urls = list.toArray(new URL[]{});
+            classLoaderMap.put(entry.getKey(), new ChildFirstClassLoader(urls, parent));
+        }
+        return unmodifiableMap(classLoaderMap);
+    }
+
+    /**
+     * Prepare processor classloaders for given job for current thread
+     *
+     * @param jobId id of the job
+     */
+    public void prepareProcessorClassLoaders(long jobId) {
+        ProcessorClassLoaderTLHolder.putAll(getProcessorClassLoaders(jobId));
+    }
+
+    private Map<String, ClassLoader> getProcessorClassLoaders(long jobId) {
+        return classLoaders.get(jobId).processorCls();
+    }
+
+    /**
+     * Clears processor classloaders from the current thread
+     */
+    public void clearProcessorClassLoaders() {
+        ProcessorClassLoaderTLHolder.remove();
+    }
+
+    /**
+     * Return processor classloader for a vertex with given name, in a job specified by the id
+     * <p>
+     * This method must be called after the classloader was created by
+     * {@link #getOrCreateClassLoader(JobConfig, long, JobPhase)} on this
+     * member.
+     *
+     * @param jobId      job id
+     * @param vertexName vertex name
+     * @return processor classloader, null if the classloader is not defined for the vertex
+     */
+    public ClassLoader getProcessorClassLoader(long jobId, String vertexName) {
+        JobClassLoaders jobClassLoaders = classLoaders.get(jobId);
+        if (jobClassLoaders != null) {
+            return jobClassLoaders.processorCl(vertexName);
+        } else {
+            throw new HazelcastException("JobClassLoaders for jobId=" + Util.idToString(jobId)
+                    + " requested, but it does not exists");
+        }
+    }
+
+    /**
+     * Try to remove and close/shutdown job classloader and any processor classloaders for given job
+     *
+     * If there are some
+     */
+    public void tryRemoveClassloadersForJob(long jobId, JobPhase phase) {
```

Comment:
I would reconsider the logging levels in this method. 
- On line 191 and 200 and 220 it should be FINEST - it's used only to evaluate issues in the implementation. 
- On line 194 it should be a WARN, i guess this is not possible unless there's some problem. 
- On line 207 it should be WARN too - it's not normal if the `close()` fails. Also, on line 206 we should catch `Throwable`, we're calling user code I think.


Also, all logs on FINE or FINEST levels should use `LoggingUtil`.